Zz fixes simplified

huitseeker opened the pull request:
This is #32 without the upstream fixes. Closes #21 #23 #27 #29

zhenfeizhang committed:
revert changes in uni-stark

zhenfeizhang committed:
fix compiling errors

zhenfeizhang committed:
address issues

zhenfeizhang committed:
fix ci

zhenfeizhang committed:
CI

zhenfeizhang committed:
fmt

zhenfeizhang committed:
remove unneccessay option for aux trace

adr1anh commented on air/src/air.rs:
```diff
@@ -224,7 +189,7 @@ pub trait ExtensionBuilder: AirBuilder<F: Field> {
     type EF: ExtensionField<Self::F>;
 
     /// Expression type over extension field elements.
-    type ExprEF: Algebra<Self::Expr> + Algebra<Self::EF>;
```
I’m not sure why this was necessary

adr1anh commented on matrix/src/horizontally_truncated.rs:
This seems to be imported from https://github.com/Plonky3/Plonky3/commit/c9d8c3e0b8b5d62e318ea8aef6e9e16da0f27f15

huitseeker replied:
Good catch, removed.

adr1anh reviewed (approved):
Looks good, I realized that we're overwriting uni-stark and still taking in some upstream changes. I don't think this is a problem, as eventually we should be able to isolate all of our own changes to only a couple of crates.

huitseeker committed:
fix: restore winter-utils dependency for goldilocks crate

huitseeker committed:
fix: remove incorrect BaseAirWithAuxTrace impl blocks from poseidon2-air

huitseeker committed:
fix: restore BaseAirWithAuxTrace trait and batch-stark files from origin/main

The 'revert changes in uni-stark' commit incorrectly removed the BaseAirWithAuxTrace trait,
and the cherry-pick conflict resolution incorrectly modified batch-stark files.
Restoring these from origin/main to fix compilation errors.

huitseeker committed:
fix: add Algebra<SymbolicExpression<F>> impl for SymbolicExpression<EF>

This is needed for the ExtensionBuilder trait bound after restoring air.rs from origin/main.

huitseeker committed:
fix: add Algebra<SymbolicExpression<F>> impl for BinomialExtensionField and update bounds

Added specific Algebra implementation for SymbolicExpression<BinomialExtensionField<F, D>>
to satisfy the ExtensionBuilder trait requirements after restoring air.rs from origin/main.
Updated where clauses in ExtensionBuilder and PermutationAirBuilder impls to include the
Algebra bound.

huitseeker committed:
fix: add backward compatibility methods and BaseAirWithAuxTrace impls

- Added constraints() method as alias for base_constraints()
- Added get_log_quotient_degree() function for backward compatibility
- Added empty BaseAirWithAuxTrace implementations for Blake3Air, KeccakAir, Poseidon2Air, and VectorizedPoseidon2Air
- Fixed trace reference types in examples/src/proofs.rs

huitseeker committed:
fix: restore batch-stark from backup and fix poseidon2-air imports

Restored batch-stark directory from backup branch to get the multi-trace implementation.
Fixed missing trait imports in poseidon2-air files after checkout.

huitseeker committed:
fix: update examples and tests for API changes

- Changed &trace to trace in prove() calls
- Added log_folding_factor to FriParameters in tests
- Changed Rc to Arc in lookup tests to match SymbolicExpression

huitseeker committed:
fix: final API compatibility fixes and formatting

- Fixed miden-prover test signatures to use &trace
- Ran cargo fmt to fix formatting issues
- All CI checks now passing (except batch-stark which is excluded)

huitseeker committed:
fix: restore batch-stark from origin/main

The backup branch had multi-trace lookup implementation that doesn't work with origin/main APIs.
Since no zzhang commits modified batch-stark/src/, using origin/main version.

huitseeker committed:
fix: restore batch-stark tests and Cargo.toml from origin/main

huitseeker committed:
fix: adapt batch-stark to multi-trace API

Update batch-stark prover and verifier to work with the multi-trace API:
- Pass aux_width=0 and num_randomness=0 to get_log_quotient_degree (batch-stark doesn't support aux traces yet)
- Remove aux_trace_local and aux_trace_next from OpenedValues construction
- Remove unused BasedVectorSpace import
- Update test to not reference aux_trace fields

batch-stark now compiles and all tests pass.

huitseeker committed:
fmt: format batch-stark code

huitseeker committed:
refactor: remove upstream SubAirBuilder and HorizontallyTruncated enhancements

Remove code from upstream Plonky3 PRs that was accidentally included:
- Remove uni-stark/src/sub_builder.rs (from upstream PR #1172 by Robin Salen)
- Remove uni-stark/tests/rc_sub_builder.rs (test for SubAirBuilder)
- Revert matrix/src/horizontally_truncated.rs to origin/main (removes new_with_range from PR #1170)

These features are not part of zzhang's multi-trace work and were only included
because they were in the 'revert changes in uni-stark' commit that was cherry-picked.
All tests still pass after removal.

huitseeker commented:
Closing in favor of https://github.com/0xmiden/p3-miden
